src/Route.php

```diff
@@ -2,56 +2,36 @@
 
 namespace Yiisoft\Router;
 
+use InvalidArgumentException;
+use LogicException;
 use Yiisoft\Router\Middleware\Callback;
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Server\MiddlewareInterface;
 use Psr\Http\Server\RequestHandlerInterface;
 
+use function current;
+use function is_callable;
+use function next;
+use function reset;
+
 /**
  * Route defines a mapping from URL to callback / name and vice versa
  */
 final class Route implements MiddlewareInterface, RequestHandlerInterface
 {
+    private string $name;
+    /** @var string[] */
+    private array $methods;
```

Comment:
In `public function __toString()` it checks for `null` which looks incorrect to me. Should check for empty array instead.